libraries/botframework-streaming/tests/NamedPipe.test.js

```diff
@@ -166,20 +164,31 @@ describe.windowsOnly('Streaming Extensions NamedPipe Library Tests', function ()
             expect(() => transport.close()).to.not.throw();
         });
 
-        // TODO: 2023-04-24 [hawo] #4462 The code today does not allows the receive() call to be rejected by reading a dead socket.
-        //                         The receive() call will be rejected IFF the socket is closed/error AFTER the receive() call.
-        it.skip('throws when reading from a dead socket', async function () {
+        it('throws when reading from a dead socket', async function () {
             const sock = new FauxSock();
             sock.destroyed = true;
             sock.connecting = false;
             sock.writable = true;
             const transport = new NamedPipeTransport(sock, 'fakeSocket5');
             expect(transport).to.be.instanceOf(NamedPipeTransport);
             expect(transport.isConnected).to.be.false;
-            (await expectEventually(transport.receive(5))).to.throw();
+            expect(() => transport.receive(5)).to.throw();
             expect(() => transport.close()).to.not.throw();
         });
 
+        it('throws when reading from a unavailable/null socket', async function () {
+            const sock = new FauxSock();
+            sock.destroyed = false;
+            sock.connecting = false;
+            sock.writable = true;
+            const transport = new NamedPipeTransport(sock, 'fakeSocket5');
+            expect(transport).to.be.instanceOf(NamedPipeTransport);
+            expect(() => transport.close()).to.not.throw();
+            expect(transport.isConnected).to.be.false;
+            expect(transport.socket).to.be.null;
+            expect(() => transport.receive(5)).to.throw();
+        });

```

Comment:
Does this work?, does this reach the error 'Cannot receive data over an unavailable/null socket.'?

I remember doing something like this, where we dont pass the socket to the NamedPipeTransport.
```suggestion
        it('throws when reading from a null socket', async function () {
            const transport = new NamedPipeTransport();
            expect(transport).to.be.instanceOf(NamedPipeTransport);
            expect(transport.isConnected).to.be.false;
            expect(() => transport.receive(5)).to.throw();
            expect(() => transport.close()).to.not.throw();
        });
```